vspace/src/test/java/edu/asu/diging/vspace/core/services/impl/ExternalLinkManagerTest.java

```diff
@@ -0,0 +1,240 @@
+        Assert.assertEquals(externalDisplayLink.getType(), savedExternalLinkDisplay1.getType());
+        Assert.assertEquals(externalDisplayLink.getExternalLink(), savedExternalLinkDisplay1.getExternalLink());
+        Mockito.verify(externalLinkDisplayRepo).save((ExternalLinkDisplay)externalDisplayLink);
+    }
+
+    @Test
+    public void test_getDisplayLink_idExists() {
+        ExternalLinkDisplay newExternalLinkDisplay = new ExternalLinkDisplay();
+        newExternalLinkDisplay.setId("EXLD001");
+        Optional<ExternalLinkDisplay> mockExternalLinkDisplay = Optional.of(newExternalLinkDisplay);
+        Mockito.when(externalLinkDisplayRepo.findById(newExternalLinkDisplay.getId())).thenReturn(mockExternalLinkDisplay);
+
+        IExternalLinkDisplay externalLinkDisplayActual = managerToTest.getDisplayLink(newExternalLinkDisplay.getId());
+        Assert.assertEquals(mockExternalLinkDisplay.get().getId(), externalLinkDisplayActual.getId());
+    }
+
+    @Test
+    public void test_getDisplayLink_idNotExists() throws Exception {
+        Mockito.when(externalLinkDisplayRepo.findById("EXLD001")).thenReturn(Optional.empty());
+        Assert.assertNull(managerToTest.getDisplayLink("EXLD001"));
+    }
+
+    @Test
+    public void test_getLink_idExists() {
+        ExternalLink newExternalLink = new ExternalLink();
+        newExternalLink.setId("EXL001");
+        Optional<ExternalLink> mockExternalLink = Optional.of(newExternalLink);
+        Mockito.when(externalLinkRepo.findById(newExternalLink.getId())).thenReturn(mockExternalLink);
+
+        IExternalLink externalLinkActual = managerToTest.getLink(newExternalLink.getId());
+        Assert.assertEquals(mockExternalLink.get().getId(), externalLinkActual.getId());
+    }
+
+    @Test
+    public void test_getLink_idNotExists() throws Exception {
+        Mockito.when(externalLinkRepo.findById("EXL001")).thenReturn(Optional.empty());
+        Assert.assertNull(managerToTest.getLink("EXL001"));
+    }
+
+    @Test
+    public void test_deleteLinkDisplayRepo_linkPresent(){
+        ISpace space = new Space();
+        space.setId("SPA001");
+        List<IExternalLinkDisplay> externalLinkDisplayList = new ArrayList<IExternalLinkDisplay>();
+        ExternalLinkDisplay externalLinkDisplay = new ExternalLinkDisplay();
+        IExternalLink externalLink = new ExternalLink();
+        externalLink.setId("EXL001");
+        externalLink.setSpace(space);
+        externalLinkDisplay.setExternalLink(externalLink);
+        externalLinkDisplayList.add(externalLinkDisplay);
+        externalLinkDisplay = new ExternalLinkDisplay();
+        externalLink = new ExternalLink();
+        externalLink.setId("EXL002");
+        externalLink.setSpace(space);
+        externalLinkDisplay.setExternalLink(externalLink);
+        externalLinkDisplayList.add(externalLinkDisplay);
+        List<IExternalLinkDisplay> externalLinkObj = externalLinkDisplayList;
```

Comment:
we want to remove the link from the list so that it does not appear on the screen after deletion. Here I am mocking that behavior by adding the data to list allowing method to remove the link from the given space.